XRTK-Core/Packages/com.xrtk.core/Runtime/Services/BoundarySystem/MixedRealityBoundarySystem.cs

```diff
@@ -28,553 +26,497 @@ public class MixedRealityBoundarySystem : BaseEventSystem, IMixedRealityBoundary
 
-                if (value && currentTrackedAreaObject.IsNull())
-                {
-                    GetTrackedAreaVisualization();
-                }
+            BoundaryDataProvider = MixedRealityToolkit.GetService<IMixedRealityBoundaryDataProvider>();
+
+            if (!Application.isPlaying || BoundaryDataProvider == null) { return; }
+
+            // Reset the bounds
+            BoundaryBounds = new Edge[0];
+            rectangularBounds = null;
+
+            // Get the boundary geometry.
+            var boundaryGeometry = new List<Vector3>(0);
+            var boundaryEdges = new List<Edge>(0);
 
-                if (!currentTrackedAreaObject.IsNull())
+            if (BoundaryDataProvider.TryGetBoundaryGeometry(ref boundaryGeometry) && boundaryGeometry.Count > 0)
+            {
+                for (int i = 0; i < boundaryGeometry.Count; i++)
                 {
-                    currentTrackedAreaObject.SetActive(value);
+                    var pointA = boundaryGeometry[i];
+                    var pointB = boundaryGeometry[(i + 1) % boundaryGeometry.Count];
+                    boundaryEdges.Add(new Edge(pointA, pointB));
                 }
 
-                RaiseBoundaryVisualizationChanged();
+                BoundaryBounds = boundaryEdges.ToArray();
+                // We always use the same seed so that from run to run, the inscribed bounds are consistent.
+                rectangularBounds = new InscribedRectangle(BoundaryBounds, Mathf.Abs("Mixed Reality Toolkit".GetHashCode()));
             }
-        }
 
-        private bool showBoundaryWalls = false;
-
-        private int trackedAreaPhysicsLayer;
+            BoundarySystemVisualizationRoot.SetActive(true);
+        }
 
         /// <inheritdoc/>
-        public int TrackedAreaPhysicsLayer
+        public override void Update()
         {
-            get
-            {
-                if (!currentTrackedAreaObject.IsNull())
-                {
-                    trackedAreaPhysicsLayer = currentTrackedAreaObject.layer;
-                }
+            base.Update();
 
-                return trackedAreaPhysicsLayer;
-            }
-            set
-            {
-                trackedAreaPhysicsLayer = value;
+            if (!Application.isPlaying || BoundaryDataProvider == null) { return; }
 
-                if (!currentTrackedAreaObject.IsNull())
+            foreach (var trackedObjectStatus in trackedObjects)
+            {
+                switch (trackedObjectStatus.Value)
                 {
-                    currentTrackedAreaObject.layer = trackedAreaPhysicsLayer;
+                    case ProximityAlert.Clear:
+                        if (!IsInsideBoundary(trackedObjectStatus.Key.transform.position))
+                        {
+                            BoundaryProximityAlert?.Invoke(trackedObjectStatus.Key, ProximityAlert.Touch);
+                        }
+                        break;
+                    case ProximityAlert.Touch:
```

Comment:
Not implemented?
Can we have comments on these if they are meant to be empty.  Else remove

Reply:
I'll finish. I must have somehow over looked this by accident.